examples/jsm/lines/LineMaterial.js

```diff
@@ -453,247 +453,181 @@ class LineMaterial extends ShaderMaterial {
 
-			},
+		if ( ! this.uniforms.linewidth ) return;
+		this.uniforms.linewidth.value = value;
 
-			dashSize: {
+	}
 
-				enumerable: true,
+	get dashed() {
 
-				get: function () {
+		return 'USE_DASH' in this.defines;
 
-					return this.uniforms.dashSize.value;
+	}
 
-				},
+	set dashed( value ) {
 
-				set: function ( value ) {
+		if ( ( value === true ) !== this.dashed ) {
 
-					this.uniforms.dashSize.value = value;
+			this.needsUpdate = true;
 
-				}
+		}
 
-			},
+		if ( value === true ) {
 
-			dashOffset: {
+			this.defines.USE_DASH = '';
 
-				enumerable: true,
+		} else {
 
-				get: function () {
+			delete this.defines.USE_DASH;
 
-					return this.uniforms.dashOffset.value;
+		}
 
-				},
+	}
 
-				set: function ( value ) {
+	get dashScale() {
 
-					this.uniforms.dashOffset.value = value;
+		return this.uniforms.dashScale.value;
 
-				}
+	}
 
-			},
+	set dashScale( value ) {
 
-			gapSize: {
+		this.uniforms.dashScale.value = value;
 
-				enumerable: true,
+	}
 
-				get: function () {
+	get dashSize() {
 
-					return this.uniforms.gapSize.value;
+		return this.uniforms.dashSize.value;
 
-				},
+	}
 
-				set: function ( value ) {
+	set dashSize( value ) {
 
-					this.uniforms.gapSize.value = value;
+		this.uniforms.dashSize.value = value;
 
-				}
+	}
 
-			},
+	get dashOffset() {
 
-			opacity: {
+		return this.uniforms.dashOffset.value;
 
-				enumerable: true,
+	}
 
-				get: function () {
+	set dashOffset( value ) {
 
-					return this.uniforms.opacity.value;
+		this.uniforms.dashOffset.value = value;
 
-				},
+	}
 
-				set: function ( value ) {
+	get gapSize() {
 
-					this.uniforms.opacity.value = value;
+		return this.uniforms.gapSize.value;
 
-				}
+	}
 
-			},
+	set gapSize( value ) {
 
-			resolution: {
+		this.uniforms.gapSize.value = value;
 
-				enumerable: true,
+	}
 
-				get: function () {
+	get opacity() {
 
-					return this.uniforms.resolution.value;
+		return this.uniforms.opacity.value;
 
-				},
+	}
 
-				set: function ( value ) {
+	set opacity( value ) {
 
-					this.uniforms.resolution.value.copy( value );
+		if ( ! this.uniforms ) return;
+		this.uniforms.opacity.value = value;
 
-				}
+	}
 
-			},
+	get resolution() {
 
-			alphaToCoverage: {
+		return this.uniforms.resolution.value;
 
-				enumerable: true,
+	}
 
-				get: function () {
+	set resolution( value ) {
 
-					return Boolean( 'USE_ALPHA_TO_COVERAGE' in this.defines );
+		this.uniforms.resolution.value.copy( value );
 
-				},
+	}
 
-				set: function ( value ) {
+	get alphaToCoverage() {
 
-					if ( Boolean( value ) !== Boolean( 'USE_ALPHA_TO_COVERAGE' in this.defines ) ) {
+		return 'USE_ALPHA_TO_COVERAGE' in this.defines;
 
-						this.needsUpdate = true;
+	}
 
-					}
+	set alphaToCoverage( value ) {
 
-					if ( value === true ) {
+		if ( ! this.defines ) return;
 
-						this.defines.USE_ALPHA_TO_COVERAGE = '';
-						this.extensions.derivatives = true;
+		if ( ( value === true ) !== this.alphaToCoverage ) {
```

Comment:
Maybe this line, too...

Reply:
This line is almost the same as was before, I just simplified it a bit.